sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -92,6 +92,11 @@ public void Start()
             started?.SetCustomProperty(AzureSdkScopeLabel, AzureSdkScopeValue);
         }
 
+        public void SetDisplayName(string displayName)
```

Comment:
Contract hints that the DisplayName is mutable right? 

//c: @lmolkova thoughts?

Reply:
it definitely is, event the underlying [DisplayName](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.displayname?view=net-7.0) property on the Activity is mutable.